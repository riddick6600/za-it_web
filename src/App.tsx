import React from "react";
import { useEffect } from "react";
import {
  BrowserRouter as Router,
  Routes,
  Route,
  useLocation,
} from "react-router-dom";
import { Main, Service, Contacts, Page404 } from "@pages";
import { Footer } from "@components";

const ScrollToTop = () => {
  const { pathname } = useLocation();

  useEffect(() => {
    window.scrollTo(0, 0);
  }, [pathname]);

  return null;
};

export const App = () => {
  return (
    <>
      <div className="App">
        <Router>
          <ScrollToTop />
          <Routes>
            <Route path="/" element={<Main />} />
            <Route path="/landing" element={<Service />} />
            <Route path="/contact" element={<Contacts />} />
            <Route path="*" element={<Page404 />} />
          </Routes>
          <Footer />
        </Router>
      </div>
    </>
  );
};
