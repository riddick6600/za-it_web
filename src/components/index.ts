export * from "./AboutUsContent";
export * from "./Features";
export * from "./MainBanner";
export * from "./WhatWeOffer";
export * from "./ServicesArea";
export * from "./ContactForm";
export * from "./ContactInfo";
export * from "./Footer";
export * from "./Navbar";
