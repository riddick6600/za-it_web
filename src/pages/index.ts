export * from "./Main";
export * from "./Service";
export * from "./Contact";
