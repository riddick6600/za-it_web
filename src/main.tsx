import React from "react";
import ReactDOM from "react-dom/client";
import { App } from "./App";

import "@public/css/bootstrap.min.css";
import "animate.css";
import "@public/css/boxicons.min.css";
import "@public/css/flaticon.css";
import "@public/css/slick.css";
// import "react-accessible-accordion/dist/fancy-example.css";
import "@public/css/style.css";
import "@public/css/responsive.css";
import "@public/css/rtl.css";

ReactDOM.createRoot(document.getElementById("root") as HTMLElement).render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);
