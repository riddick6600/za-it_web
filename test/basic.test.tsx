import React from "react";
import renderer from "react-test-renderer";
import { App } from "../src/App";

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();
  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);
  return result as renderer.ReactTestRendererJSON;
}

test("Render Component", () => {
  const component = renderer.create(<App></App>);
  let tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
