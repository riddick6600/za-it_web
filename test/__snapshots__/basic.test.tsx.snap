// Vitest Snapshot v1

exports[`Render Component 1`] = `
<div
  className="App"
>
  <header
    className="headroom"
    id="header"
  >
    <div
      className="startp-nav"
    >
      <div
        className="container"
      >
        <nav
          className="navbar navbar-expand-md navbar-light"
        >
          <a
            className="navbar-brand"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="logo"
              src="/images/logo.svg"
            />
          </a>
        </nav>
      </div>
    </div>
  </header>
  <div
    className="ml-main-section main-banner"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-6 col-md-12"
        >
          <div
            className="hero-content"
          >
            <h1>
              Разворачиваем инфраструктуры
            </h1>
            <h2>
              Администрируем серверы 
              <br />
              Мониторим системы 24/7
            </h2>
            <p>
              За лучшим ИТ для вашего бизнеса — к нам
            </p>
            <a
              className="btn btn-secondary"
              href="/contact"
              onClick={[Function]}
            >
              Оставить заявку
            </a>
          </div>
        </div>
        <div
          className="col-lg-6 col-md-12"
        >
          <div
            className="ml-banner-image"
          >
            <img
              alt="image"
              className="animate__animated animate__fadeIn animate__delay-2s"
              src="/images/ml-banner-image/one.png"
            />
            <img
              alt="image"
              className="animate__animated animate__fadeInUp animate__delay-1s"
              src="/images/ml-banner-image/two.png"
            />
            <img
              alt="image"
              className="animate__animated animate__fadeInUp animate__delay-0.5s"
              src="/images/ml-banner-image/three.png"
            />
            <img
              alt="image"
              className="animate__animated animate__fadeInUp animate__delay-2s"
              src="/images/ml-banner-image/four.png"
            />
            <img
              alt="image"
              className="animate__animated animate__fadeInDown animate__delay-2.5s"
              src="/images/ml-banner-image/five.png"
            />
            <img
              alt="image"
              className="animate__animated animate__fadeInDown animate__delay-1.9s"
              src="/images/ml-banner-image/six.png"
            />
            <img
              alt="image"
              className="animate__animated animate__fadeInDown animate__delay-2.1s"
              src="/images/ml-banner-image/seven.png"
            />
            <img
              alt="image"
              className="animate__animated animate__fadeInDown animate__delay-1.9s"
              src="/images/ml-banner-image/eight.png"
            />
            <img
              alt="image"
              className="animate__animated animate__fadeInDown animate__delay-1.7s"
              src="/images/ml-banner-image/nine.png"
            />
            <img
              alt="image"
              className="animate__animated animate__fadeInDown animate__delay-1.5s"
              src="/images/ml-banner-image/ten.png"
            />
            <img
              alt="image"
              className="animate__animated animate__fadeInUp animate__delay-0.4s"
              src="/images/ml-banner-image/eleven.png"
            />
            <img
              alt="image"
              className="animate__animated animate__fadeInUp animate__delay-0.5s"
              src="/images/ml-banner-image/ml-main-pic.png"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="shape1"
    >
      <img
        alt="shape"
        src="/images/shape1.png"
      />
    </div>
    <div
      className="shape3"
    >
      <img
        alt="shape"
        src="/images/shape3.svg"
      />
    </div>
    <div
      className="shape4"
    >
      <img
        alt="shape"
        src="/images/shape4.svg"
      />
    </div>
    <div
      className="shape6 rotateme"
    >
      <img
        alt="shape"
        src="/images/shape4.svg"
      />
    </div>
    <div
      className="shape7"
    >
      <img
        alt="shape"
        src="/images/shape4.svg"
      />
    </div>
    <div
      className="shape8 rotateme"
    >
      <img
        alt="shape"
        src="/images/shape2.svg"
      />
    </div>
  </div>
  <div
    className="solutions-area"
  >
    <div
      className="container"
    >
      <div
        className="section-title"
      >
        <h2>
          Наши решения
        </h2>
        <div
          className="bar"
        />
      </div>
      <div
        className="row justify-content-center"
      >
        <div
          className="col-lg-3 col-sm-6 col-md-6"
        >
          <div
            className="single-solutions-box"
          >
            <div
              className="icon"
            >
              <img
                alt="image"
                src="/images/icon4.png"
              />
            </div>
            <h3>
              ПО и оборудование
            </h3>
            <p>
              Подберем, установим ПО и подготовим оборудование
            </p>
          </div>
        </div>
        <div
          className="col-lg-3 col-sm-6 col-md-6"
        >
          <div
            className="single-solutions-box"
          >
            <div
              className="icon"
            >
              <img
                alt="image"
                src="/images/icon5.png"
              />
            </div>
            <h3>
              Сети и автоматизация
            </h3>
            <p>
              Развернем и настроим серверы, Wi-Fi, ЛВС, видеоконференцсвязь. Устраним рутину развертывания рабочих мест
            </p>
          </div>
        </div>
        <div
          className="col-lg-3 col-sm-6 col-md-6"
        >
          <div
            className="single-solutions-box"
          >
            <div
              className="icon"
            >
              <img
                alt="image"
                src="/images/icon6.png"
              />
            </div>
            <h3>
              Безопасность и видеонаблюдение
            </h3>
            <p>
              Установим СКУД, обеспечим работу датчиков
            </p>
          </div>
        </div>
        <div
          className="col-lg-3 col-sm-6 col-md-6"
        >
          <div
            className="single-solutions-box"
          >
            <div
              className="icon"
            >
              <img
                alt="image"
                src="/images/icon7.png"
              />
            </div>
            <h3>
              Круглосуточная поддержка
            </h3>
            <p>
              Любая ИТ-поддержка 24/7 онлайн и с выездом
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="shape1"
    >
      <img
        alt="shape"
        src="/images/shape1.png"
      />
    </div>
    <div
      className="shape2 rotateme"
    >
      <img
        alt="shape"
        src="/images/shape2.svg"
      />
    </div>
    <div
      className="shape3"
    >
      <img
        alt="shape"
        src="/images/shape3.svg"
      />
    </div>
    <div
      className="shape4"
    >
      <img
        alt="shape"
        src="/images/shape4.svg"
      />
    </div>
    <div
      className="shape6 rotateme"
    >
      <img
        alt="shape"
        src="/images/shape4.svg"
      />
    </div>
    <div
      className="shape7"
    >
      <img
        alt="shape"
        src="/images/shape4.svg"
      />
    </div>
    <div
      className="shape8 rotateme"
    >
      <img
        alt="shape"
        src="/images/shape2.svg"
      />
    </div>
  </div>
  <div
    className="about-area"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row align-items-center"
      >
        <div
          className="col-lg-6 col-md-12"
        >
          <div
            className="ml-about-img"
          >
            <img
              alt="image"
              src="/images/about4.png"
            />
          </div>
        </div>
        <div
          className="col-lg-6 col-md-12"
        >
          <div
            className="ml-about-content"
          >
            <h3>
              ПО и Оборудование
            </h3>
            <div
              className="bar"
            />
            <p>
              Инсталляция серверного и пользовательского программного обеспечения и оборудования
            </p>
            <p>
              Установка ОС Windows, Linux
            </p>
            <p>
              Сборка компьютеров
            </p>
            <p>
              Подготовка и ремонт периферийной техники
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="shape1"
    >
      <img
        alt="shape"
        src="/images/shape1.png"
      />
    </div>
    <div
      className="shape2 rotateme"
    >
      <img
        alt="shape"
        src="/images/shape2.svg"
      />
    </div>
    <div
      className="shape3"
    >
      <img
        alt="shape"
        src="/images/shape3.svg"
      />
    </div>
    <div
      className="shape4"
    >
      <img
        alt="shape"
        src="/images/shape4.svg"
      />
    </div>
    <div
      className="shape7"
    >
      <img
        alt="shape"
        src="/images/shape4.svg"
      />
    </div>
    <div
      className="shape8 rotateme"
    >
      <img
        alt="shape"
        src="/images/shape2.svg"
      />
    </div>
  </div>
  <div
    className="iot-features-area ptb-50 bg-f7fafd"
  >
    <div
      className="container"
    >
      <div
        className="row align-items-center"
      >
        <div
          className="col-lg-6 iot-features-content"
        >
          <h3>
            Сетевая инфраструктура и автоматизация
          </h3>
          <p>
            Установка серверного оборудования
          </p>
          <p>
            Подключение Wi-Fi и проводного доступа
          </p>
          <p>
            Установка и настройка сетевого оборудования и служб
          </p>
          <p>
            Автоматизация, мониторинг серверных служб и ролей
          </p>
        </div>
        <div
          className="col-lg-6 iot-features-image"
        >
          <img
            alt="image"
            className="animate__animated animate__fadeInUp animate__delay-0.6s"
            src="/images/iot-features-image/iot-feature-image1.png"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="iot-features-area ptb-80"
  >
    <div
      className="container"
    >
      <div
        className="row align-items-center"
      >
        <div
          className="col-lg-6 iot-features-image"
        >
          <img
            alt="image"
            className="animate__animated animate__fadeInUp animate__delay-0.6s"
            src="/images/iot-features-image/iot-feature-image2.png"
          />
        </div>
        <div
          className="col-lg-6 iot-features-content"
        >
          <h3>
            Безопасность и видеонаблюдение
          </h3>
          <p>
            Установка СКУД
          </p>
          <p>
            Настройка пропускных точек, отладка датчиков контроля
          </p>
          <p>
            Установка видеонаблюдения
          </p>
          <p>
            Настройка автоматической системы оповещения
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="services-area ptb-80 bg-f7fafd"
  >
    <div
      className="container"
    >
      <div
        className="row justify-content-center align-items-center"
      >
        <div
          className="col-lg-6 col-md-12 services-content"
        >
          <div
            className="section-title"
          >
            <h2>
              Поддержка
            </h2>
            <div
              className="bar"
            />
            <p>
              Service Desk
            </p>
            <p>
              Техподдержка 1С
            </p>
            <p>
              Выезды на объект
            </p>
            <br />
            <a
              className="btn btn-secondary"
              href="/contact"
              onClick={[Function]}
            >
              Оставить заявку
            </a>
          </div>
        </div>
        <div
          className="col-lg-6 col-md-12 services-right-image"
        >
          <img
            alt="book-self"
            className="animate__animated animate__fadeInDown animate__delay-0.2s"
            src="/images/services-right-image/book-self.png"
          />
          <img
            alt="box"
            className="animate__animated animate__fadeInUp animate__delay-0.2s"
            data-wow-delay="0.6s"
            src="/images/services-right-image/box.png"
          />
          <img
            alt="chair"
            className="animate__animated animate__fadeInLeft animate__delay-0.2s"
            src="/images/services-right-image/chair.png"
          />
          <img
            alt="cloud"
            className="animate__animated animate__zoomIn animate__delay-0.2s"
            src="/images/services-right-image/cloud.png"
          />
          <img
            alt="cup"
            className="animate__animated animate__bounceIn animate__delay-0.2s"
            src="/images/services-right-image/cup.png"
          />
          <img
            alt="flower"
            className="animate__animated animate__fadeInDown animate__delay-0.2s"
            src="/images/services-right-image/flower-top.png"
          />
          <img
            alt="head-phone"
            className="animate__animated animate__zoomIn animate__delay-0.2s"
            src="/images/services-right-image/head-phone.png"
          />
          <img
            alt="monitor"
            className="animate__animated animate__fadeInUp animate__delay-0.2s"
            src="/images/services-right-image/monitor.png"
          />
          <img
            alt="mug"
            className="animate__animated animate__rotateIn animate__delay-0.2s"
            src="/images/services-right-image/mug.png"
          />
          <img
            alt="table"
            className="animate__animated animate__fadeInUp animate__delay-0.2s"
            src="/images/services-right-image/table.png"
          />
          <img
            alt="tissue"
            className="animate__animated animate__zoomIn animate__delay-0.2s"
            src="/images/services-right-image/tissue.png"
          />
          <img
            alt="water-bottle"
            className="animate__animated animate__fadeInUp animate__delay-0.2s"
            src="/images/services-right-image/water-bottle.png"
          />
          <img
            alt="wifi"
            className="animate__animated animate__fadeInLeft animate__delay-0.2s"
            data-wow-delay="0.6s"
            src="/images/services-right-image/wifi.png"
          />
          <img
            alt="shape"
            className="bg-image rotateme"
            src="/images/services-right-image/cercle-shape.png"
          />
          <img
            alt="main-pic"
            className="animate__animated animate__fadeInUp animate__delay-0.2s"
            src="/images/services-right-image/service-right-main-pic.png"
          />
        </div>
      </div>
    </div>
  </div>
  <footer
    className="footer-area"
  >
    <div
      className="container"
    >
      <div>
        <img
          alt="map"
          className="map"
          src="/images/map.png"
        />
      </div>
      <div
        className="row trs_center"
      >
        <div
          className="col-lg-3 col-md-6"
        >
          <div
            className="single-footer-widget"
          >
            <div
              className="logo"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                <img
                  alt="logo"
                  src="/images/logo.svg"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="col-lg-3 col-md-6"
        >
          <div
            className="single-footer-widget"
          >
            <a
              href="/contact"
              onClick={[Function]}
            >
              <h3>
                Контакты
              </h3>
            </a>
            <div>
              <a
                href="mailto:admin@za-it.ru"
              >
                admin@za-it.ru
              </a>
            </div>
            <br />
            <div>
              <a
                href="tel:+7 (967) 555-41-34"
              >
                +7 (967) 555-41-34
              </a>
            </div>
          </div>
        </div>
        <div
          className="col-lg-12 col-md-12"
        >
          <div
            className="copyright-area"
          >
            <p>
              Copyright © 2021 
              za-it
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="shape1"
    >
      <img
        alt="shape"
        src="/images/shape1.png"
      />
    </div>
    <div
      className="shape8 rotateme"
    >
      <img
        alt="shape"
        src="/images/shape2.svg"
      />
    </div>
  </footer>
</div>
`;
